src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -315,51 +303,68 @@ private List<ModificationInfos> execCreateSwitchOnLine(NetworkStoreListener list
                     .withValue("id", lineId)
                     .withSeverity(TypedValue.INFO_SEVERITY)
                     .build());
-            }
-
-                // add the branch status modification entity to the listener
-                listener.storeBranchStatusModification(lineId, BranchStatusModificationInfos.ActionType.SWITCH_ON);
-            }, MODIFICATION_ERROR, reportUuid, reporter, subReporter
+            },
+            () -> listener.storeBranchStatusModification(lineId, BranchStatusModificationInfos.ActionType.SWITCH_ON),
+            MODIFICATION_ERROR, reportUuid, reporter, subReporter
         );
     }
 
     public void deleteModificationGroup(UUID groupUuid, boolean errorOnGroupNotFound) {
         networkModificationRepository.deleteModificationGroup(groupUuid, errorOnGroupNotFound);
     }
 
-    public List<ModificationInfos> doAction(NetworkStoreListener listener, Runnable action,
-                                            NetworkModificationException.Type typeIfError,
-                                            UUID reportUuid, ReporterModel reporter,
-                                            Reporter subReporter) {
+    private NetworkModificationException handleException(NetworkModificationException.Type typeIfError, Reporter subReporter, Exception e) {
+        NetworkModificationException networkModificationException;
+        networkModificationException = e instanceof NetworkModificationException ? (NetworkModificationException) e : new NetworkModificationException(typeIfError, e);
+        boolean isApplicationException = PowsyblException.class.isAssignableFrom(e.getClass());
+        if (!isApplicationException && LOGGER.isErrorEnabled()) {
+            LOGGER.error(e.toString(), e);
+        }
+        String errorMessage = isApplicationException ? e.getMessage() : "Technical error: " + e;
+        subReporter.report(Report.builder()
+            .withKey(typeIfError.name())
+            .withDefaultMessage(errorMessage)
+            .withSeverity(TypedValue.ERROR_SEVERITY)
+            .build());
+        return networkModificationException;
+    }
+
+    private List<ModificationInfos> doAction(NetworkStoreListener listener, Runnable action, Runnable postAction,
```

Comment:
since it's the heart of this micro service, I think it could save time for future developers to comment those steps to describe what is expected from each of them, and what each of those booleans mean (isApplyModification, isbuild)
I am myself a bit confused

Reply:
Yes,
This will be done in
https://dev.azure.com/GridSuite/MVP/_workitems/edit/1721